services/livesync-service-base.ts

```diff
@@ -162,62 +162,72 @@ class LiveSyncServiceBase implements ILiveSyncServiceBase {
-					if (deviceAppData.isLiveSyncSupported().wait()) {
-						let platformLiveSyncService = this.resolvePlatformLiveSyncService(platform, device);
+	public getSyncAction(data: ILiveSyncData, filesToSync?: string[], deviceFilesAction?: (device: Mobile.IDevice, localToDevicePaths: Mobile.ILocalToDevicePathData[]) => IFuture<void>, liveSyncOptions?: { isForCompanionApp: boolean }): (device: Mobile.IDevice) => IFuture<void> {
+		let appIdentifier = data.appIdentifier;
+		let platform = data.platform;
+		let projectFilesPath = data.projectFilesPath;
+
+		let packageFilePath: string = null;
+
+		let action = (device: Mobile.IDevice): IFuture<void> => {
+			return (() => {
+				let shouldRefreshApplication = true;
+				let deviceAppData = this.$deviceAppDataFactory.create(appIdentifier, this.$mobileHelper.normalizePlatformName(platform), device, liveSyncOptions);
+				if (deviceAppData.isLiveSyncSupported().wait()) {
+					let platformLiveSyncService = this.resolvePlatformLiveSyncService(platform, device);
+
+					if (platformLiveSyncService.beforeLiveSyncAction) {
+						platformLiveSyncService.beforeLiveSyncAction(deviceAppData).wait();
+					}
 
-						if (platformLiveSyncService.beforeLiveSyncAction) {
-							platformLiveSyncService.beforeLiveSyncAction(deviceAppData).wait();
+					// Not installed application
+					if (!device.applicationManager.isApplicationInstalled(appIdentifier).wait() && !this.$options.companion) {
+						this.$logger.warn(`The application with id "${appIdentifier}" is not installed on device with identifier ${device.deviceInfo.identifier}.`);
+						if (!packageFilePath) {
+							packageFilePath = this.$liveSyncProvider.buildForDevice(device).wait();
 						}
-
-						// Not installed application
-						if (!device.applicationManager.isApplicationInstalled(appIdentifier).wait() && !this.$options.companion) {
-							this.$logger.warn(`The application with id "${appIdentifier}" is not installed on device with identifier ${device.deviceInfo.identifier}.`);
-							if (!packageFilePath) {
-								packageFilePath = this.$liveSyncProvider.buildForDevice(device).wait();
-							}
-							device.applicationManager.reinstallApplication(appIdentifier, packageFilePath).wait();
-							if (device.applicationManager.canStartApplication()) {
-								device.applicationManager.startApplication(appIdentifier).wait();
-							}
-
-							if (platformLiveSyncService.afterInstallApplicationAction) {
-								let localToDevicePaths = this.$projectFilesManager.createLocalToDevicePaths(deviceAppData, projectFilesPath, filesToSync, data.excludedProjectDirsAndFiles);
-								platformLiveSyncService.afterInstallApplicationAction(deviceAppData, localToDevicePaths).wait();
-							}
-							shouldRefreshApplication = false;
+						device.applicationManager.reinstallApplication(appIdentifier, packageFilePath).wait();
```

Comment:
Change `device.applicationManager.reinstallApplication` to `device.applicationManager.installApplication` because the application is not installed yet and the uninstall in the `reinstallApplication` method can fail.